File: Python/Problem Solving/CodeForces/Others/Perfection.py


array = [int (x) for x in input().split()]


def check(x):
    sum=0
    div=int(x/2)
    for i in range(1,div+1):
        if x%i==0:
            sum+=i;
    return sum
print("PERFECTION OUTPUT")
for i in range(len(array)-1):
    x= array[i];
    if(check(x)==x):
        print("%5d  PERFECT"%x);
    elif (check(x)<x):
        print("%5d  DEFICIENT"%x);
    elif(check(x)>x):
        print("%5d  ABUNDANT"%x)
print("END OF OUTPUT")

File: Java/Problem Solving/Twinkle Twinkle Little Star/src/UVA/Perfection.java
package UVA;

import Fast_IO.InputReader;
import java.io.PrintWriter;

public class Perfection {
    public void solve(int testNumber, InputReader in, PrintWriter out) {
        out.print("PERFECTION OUTPUT\n");
        while (in.hasNext()){
            int x = in.nextInt();
            if(x==0)
                break;

            if(check(x)==x)
                out.println(String.format("%5d  PERFECT",x));
            else if (check(x)<x)
            out.println(String.format("%5d  DEFICIENT",x));
            else if(check(x)>x)
            out.println(String.format("%5d  ABUNDANT",x));


        }
        out.println("END OF OUTPUT");
    }
    static  int check (int x ){
        int sum=0;
        for (int i = 1; i <x ; i++) {
            if(x%i==0)
                sum+=i;
        }
        return sum;
    }
}
